pkgs/build-support/docker/default.nix

```
@@ -513,6 +513,8 @@ rec {
       keepContentsDirlinks ? false
     , # Docker config; e.g. what command to run on the container.
       config ? null
+    , # Image architecture, defaultArch will be used if null
+      arch ? defaultArch
```

```suggestion
    , # Image architecture, defaults to the architecture of the `hostPlatform` when unset
      architecture ? defaultArch
```

Please add a test case to `nixos/tests/docker-tools.nix` and update `doc/builders/images/dockertools.section.md`.

---

I just literally copied the approach that was used here: https://github.com/NixOS/nixpkgs/blob/97f6e8b3e63837ebd54e5401c1be2a3ba4d44638/pkgs/build-support/docker/default.nix#L103 hence wanting to keep the same variable names.

---

`defaultArch` is an internal `let` binding - a truly internal name. I don't expect such names to be particularly good ones.

`arch` on the other hand would be an exposed name, so it has to fit its domain, which is the OCI spec, which uses `architecture`. The `defaultArch` and `GOARCH` names are irrelevant.

Please change it.

---

I'll change it. But by the same logic it should also be changed for `pullImage`.

---

@roberth as per your request, done.